reflect/src/main/java/dagger/reflect/LinkedMapBinding.java

```diff
@@ -17,4 +18,28 @@
     }
     return map;
   }
+
+  Binding.LinkedBinding<Map<K, Provider<V>>> asLinkedMapProvidesBinding() {
+    Map<K, LinkedBinding<Provider<V>>> map = new LinkedHashMap<>(entryBindings.size());
+    for (Map.Entry<K, LinkedBinding<V>> elementBinding : entryBindings.entrySet()) {
+      map.put(elementBinding.getKey(), new LinkedProviderBindingWrapper<>(elementBinding.getValue()));
+    }
+    return new LinkedMapProviderBinding<>(map);
+  }
+
+  private final static class LinkedMapProviderBinding<K, V> extends Binding.LinkedBinding<Map<K, Provider<V>>> {
```

Comment:
did a little refactoring to make this an implementation detail instead of a separate class. does that make sense?